test/e2e2/conftest.py

```diff
@@ -281,6 +281,25 @@ def get_key_params(self, domain_name: str, rr_type: str) -> list:
     def assert_rrsets(self, rrsets: dict, init_rrsets: dict = None) -> None:
         init_rrsets = init_rrsets or {}
 
+        # Build expectation from rrsets and init_rrsets
+        rrsets_to_check = {
+            (subname, qtype): rrsets.get((subname, qtype)) or init_rrsets[(subname, qtype)]
+            for (subname, qtype) in rrsets.keys() | init_rrsets.keys()
```

Comment:
This method [has been (and should) be testing for records from `rrsets`](https://github.com/desec-io/desec-stack/blob/0403b4186c740d235e1bae2c63b26bf5a73089b8/test/e2e2/conftest.py#L349-L350), but not generally test records from `init_rrsets`. Only [if the `rrsets` dictionary has a `None` value](https://github.com/desec-io/desec-stack/blob/0403b4186c740d235e1bae2c63b26bf5a73089b8/test/e2e2/spec/test_api_rrset.py#L28), it should fall back to the values from `init_rrsets`. It should not otherwise test unrelated records from `init_rrsets`.

The iterator in the end should thus be just `rrsets.keys()`, not the union with the keys of `init_rrsets`.

Correct provisioning of (all) [`init_rrsets` is tested during initialization of the fixture](https://github.com/desec-io/desec-stack/blob/0403b4186c740d235e1bae2c63b26bf5a73089b8/test/e2e2/conftest.py#L364).

Reply:
fixed